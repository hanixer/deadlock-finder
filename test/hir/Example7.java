package showcase;

/**
 * Using break and continue.
 */
public class Example7 {
    public static void func1(int a, int b) {        
        int c = 0;
        while (a < b) {
            if (a == 2)
                continue;
            if (b == 2 * a)
                break;
            c = c - a;
            b = b * 1;
        }
    }
}
