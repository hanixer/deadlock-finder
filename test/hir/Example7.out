func showcase.Example7.func1(a: int, b: int): void {
  var c: int = 0
  loop {
    var t~1: boolean = a<b
    var t~2: boolean = !t~1
    if (t~2) break
    var t~3: boolean = a==2
    if (t~3) {
      continue
    }
    var t~4: int = 2*a
    var t~5: boolean = b==t~4
    if (t~5) {
      break
    }
    c = c-a
    b = b*1
  }
}
